server/container/guice/guice-common/src/test/java/org/apache/james/JamesServerExtensionBuilder.java

```diff
@@ -83,9 +83,10 @@ public JamesServerExtensionBuilder disableAutoStart() {
     public JamesServerExtension build() {
         Preconditions.checkNotNull(server);
         ConfigurationProvider configuration = this.configuration.orElse(defaultConfigurationProvider());
+        JamesServerExtension.AwaitCondition awaitCondition = () -> extensions.build().forEach(GuiceModuleTestExtension::await);
```

Comment:
If we don't support specifying custom `awaitCondtion`. Then `awaitCondition` could be directly created in the `JamesServerExtension` constructor instead of from the builder => for simplify the constructor regarding to the number of parameters. E.g:
```java
    JamesServerExtension(RegistrableExtension registrableExtension, ThrowingFunction<File, GuiceJamesServer> serverSupplier, boolean autoStart) {
        ....
        this.awaitCondition = () -> extensions.build().forEach(RegistrableExtension::await);
    }	    
}
```

Reply:
No because it is done from the GuiceTestModule, not from the extensions.

Furthermore, anyway, moving init logic from a constructor to its builder seems like a good move to me!